sklearn/utils/estimator_checks.py

```diff
@@ -3899,6 +3900,40 @@ def _enforce_estimator_tags_X(estimator, X, X_test=None, kernel=linear_kernel):
     return X_res
 
 
+@ignore_warnings(category=FutureWarning)
+def check_positive_only_tag_during_fit(name, estimator_orig):
+    """Test that the estimator correctly sets the tags.input_tags.positive_only
+
+    If the tag is False, the estimator should accept negative input regardless of the
+    tags.input_tags.pairwise flag.
+    """
+    estimator = clone(estimator_orig)
+    tags = get_tags(estimator)
+
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    y = _enforce_estimator_tags_y(estimator, y)
+    set_random_state(estimator, 0)
+    X = _enforce_estimator_tags_X(estimator_orig, X)
```

Comment:
why is it estimator for y and estimator_orig for X ? Shouldn't it be estimator for both ?